lib/src/paginated_search_result.dart

```diff
@@ -33,8 +33,10 @@ class PaginatedSearchResult extends Searcheable {
   /// Total number of pages
   final int? totalPages;
 
-  factory PaginatedSearchResult.fromMap(Map<String, Object?> map,
-      {String? indexUid}) {
+  static PaginatedSearchResult<Map<String, Object?>> fromMap(
+    Map<String, Object?> map, {
+    String? indexUid,
+  }) {
     return PaginatedSearchResult(
```

Comment:
Don't you have to make a modification here as well?

Reply:
nope, since dart can infer the correct type